### PR TITLE
OSS bug fixes

### PR DESCRIPTION
**Pull request by jatingaur18**
## Description

Fixed bugs 
Slack - `find_slack_burst_record_by_ts` query 
UI - fixed record type to show record type instead of file type 
md parser - image finder regex
Docling parser - catch `ConversionError` for unsupported file formats (e.g. macOS `._` files)

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [ ] Unit tests pass
- [ ] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [ ] **Search capabilities** are working correctly
- [ ] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [ ] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
image

**After:**
image

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

## Summary by CodeRabbit

* **New Features**
  * Expanded knowledge-base support for diverse record types, including drives, databases, mail, collaboration tools, projects, and meetings.
  * Added localized labels for file and artifact records across supported languages.
  * Slack burst records can now be retrieved from the graph database.

* **Bug Fixes**
  * Improved Markdown image parsing for nested alt-text brackets and data URLs containing spaces.
  * Document conversion errors now provide clearer, more specific feedback.
  * Mail, group-mail, URL filters, and record-type metadata display more accurately.

### CODE REVIEW

File: frontend/lib/utils/record-type-labels.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export const RECORD_TYPES = [
```
[the remark this answers is not in the document]
Comment (jatingaur18) [reply frontend/lib/utils/record-type-labels.ts:1]: The codebase convention only uses 'use client' on .tsx component files and .ts files that use React hooks

File: frontend/app/(main)/record/components/record-metadata-panel.tsx [hunk 2]
```diff
@@ -177,22 +177,12 @@ function shouldShowIndexingReasonTooltip(status: string | undefined, reason: str
...
-  if (recordType === 'TICKET') return t('recordView.labels.typeTicket');
-  if (recordType === 'WEBPAGE') return t('recordView.labels.typeWebpage');
-  if (recordType === 'MESSAGE') return t('recordView.labels.typeMessage');
-  return recordType;
+function isMailRecordType(recordType: string): boolean {
+  return recordType === 'EMAIL' || recordType === 'MAIL' || recordType === 'GROUP_MAIL';
+}
+
+function formatDisplayType(record: RecordDetailsResponse['record']): string {
+  return record.recordType;
```
[the remark this answers is not in the document]
Comment (jatingaur18) [reply frontend/app/(main)/record/components/record-metadata-panel.tsx]: resolved

File: backend/python/app/modules/parsers/markdown/markdown_to_blocks.py [hunk 3]
```diff
@@ -25,7 +25,11 @@
 )
 from app.modules.parsers.text_splitting import split_long_text
 
-_MD_IMAGE_RE = re.compile(r'!\[([^\]]*)\]\(([^\s)]+)(?:\s+"[^"]*")?\)')
+_MD_IMAGE_RE = re.compile(
+    r'!\[((?:[^\[\]]|\[[^\]]*\])*)\]\('
+    r'(data:(?:[^"\s)]+|\s+(?!"[^"]*"\s*\))|"[^")]*)*|[^\s)]+)'
```
[the remark this answers is not in the document]
Comment (jatingaur18) [reply backend/python/app/modules/parsers/markdown/markdown_to_blocks.py]: Fixed

File: backend/python/app/modules/parsers/markdown/markdown_to_blocks.py [hunk 4]
```diff
@@ -25,7 +25,11 @@
 )
 from app.modules.parsers.text_splitting import split_long_text
 
-_MD_IMAGE_RE = re.compile(r'!\[([^\]]*)\]\(([^\s)]+)(?:\s+"[^"]*")?\)')
+_MD_IMAGE_RE = re.compile(
+    r'!\[((?:[^\[\]]|\[[^\]]*\])*)\]\('
+    r'(data:(?:[^"\s)]|\s+(?!"[^"]*"\s*\))|"[^")]*)*|[^\s)]+)'
```
[the remark this answers is not in the document]
Comment (jatingaur18) [reply backend/python/app/modules/parsers/markdown/markdown_to_blocks.py:30]: Fixed

**Review by abhishekgml1991 — APPROVED** [review]
